test/e2e/framework/addon/vault/setup.go

```diff
@@ -343,61 +319,81 @@ func (v *VaultInitializer) Setup() error {
 }
 
 func (v *VaultInitializer) Clean() error {
-	if err := v.client.Sys().Unmount("/" + v.intermediateMount); err != nil {
+	ctx := context.Background()
+
+	if _, err := v.client.System.MountsDisableSecretsEngine(ctx, "/"+v.intermediateMount); err != nil {
 		return fmt.Errorf("unable to unmount %v: %v", v.intermediateMount, err)
 	}
-	if err := v.client.Sys().Unmount("/" + v.rootMount); err != nil {
+	if _, err := v.client.System.MountsDisableSecretsEngine(ctx, "/"+v.rootMount); err != nil {
 		return fmt.Errorf("unable to unmount %v: %v", v.rootMount, err)
 	}
 
 	return nil
 }
 
 func (v *VaultInitializer) CreateAppRole() (string, string, error) {
+	ctx := context.Background()
+
 	// create policy
 	policy := fmt.Sprintf(`path "%s" { capabilities = [ "create", "update" ] }`, v.IntermediateSignPath())
-	err := v.client.Sys().PutPolicy(v.role, policy)
+	_, err := v.client.System.PoliciesWriteAclPolicy(
+		ctx,
+		v.role,
+		schema.PoliciesWriteAclPolicyRequest{
+			Policy: policy,
+		},
+	)
 	if err != nil {
 		return "", "", fmt.Errorf("error creating policy: %s", err.Error())
 	}
 
 	// # create approle
-	params := map[string]string{
-		"period":   "24h",
-		"policies": v.role,
-	}
-
-	baseUrl := path.Join("/v1", "auth", v.appRoleAuthPath, "role", v.role)
-	_, err = v.callVault("POST", baseUrl, "", params)
+	_, err = v.client.Auth.AppRoleWriteRole(
+		ctx,
+		v.role,
+		schema.AppRoleWriteRoleRequest{
+			Period:   "24h",
+			Policies: []string{v.role},
+		},
+		vault.WithMountPath(v.appRoleAuthPath),
+	)
 	if err != nil {
 		return "", "", fmt.Errorf("error creating approle: %s", err.Error())
 	}
 
 	// # read the role-id
-	url := path.Join(baseUrl, "role-id")
-	roleId, err := v.callVault("GET", url, "role_id", map[string]string{})
+	respRoleId, err := v.client.Auth.AppRoleReadRoleId(
+		ctx,
+		v.role,
+		vault.WithMountPath(v.appRoleAuthPath),
+	)
 	if err != nil {
 		return "", "", fmt.Errorf("error reading role_id: %s", err.Error())
 	}
 
 	// # read the secret-id
-	url = path.Join(baseUrl, "secret-id")
-	secretId, err := v.callVault("POST", url, "secret_id", map[string]string{})
```

Comment:
Notice that the original comment is misleading. These lines are writing (POSTing) to the secret-id endpoint.

Reply:
ah, good catch. Vault seems to use POST-as-GET, is that what is happening there? 

The error message below also refers to "reading":

```go
fmt.Errorf("error reading secret_id: %s", ...)
```